discovery-provider/src/tasks/index_hourly_play_counts.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import time
+from typing import List
+from sqlalchemy import func, desc, text
+from src.models.models import HourlyPlayCounts
+from src.utils.update_indexing_checkpoints import UPDATE_INDEXING_CHECKPOINTS_QUERY
+from src.tasks.celery_app import celery
+from src.models import IndexingCheckpoints, Play
+
+logger = logging.getLogger(__name__)
+
+HOURLY_PLAY_COUNTS_TABLE_NAME = "hourly_play_counts"
+
+UPSERT_HOURLY_PLAY_COUNTS_QUERY = """
+    INSERT INTO hourly_play_counts (hourly_timestamp, play_count)
+    VALUES(:hourly_timestamp, :play_count)
+    ON CONFLICT (hourly_timestamp)
+    DO UPDATE SET play_count = hourly_play_counts.play_count + EXCLUDED.play_count;
+    """
+
+def _index_hourly_play_counts(session):
+    # get checkpoints
+    prev_id_checkpoint = (session.query(IndexingCheckpoints.last_checkpoint)
+        .filter(IndexingCheckpoints.tablename == HOURLY_PLAY_COUNTS_TABLE_NAME)
+    ).scalar()
+
+    if not prev_id_checkpoint:
+        prev_id_checkpoint = 0
+
+    new_id_checkpoint = (
+        session.query(Play.id)
```

Comment:
`SELECT MAX(Play.id) from Plays` might be a slightly cleaner way to do this

Reply:
yeah definitely 👍